i18n(pt-BR): Updated `guides/server-side-rendering.mdx`, `guides/upgrade-to/v2.mdx`, `reference/errors/static-redirect-not-available.mdx`, and a bit of `reference/api-reference.mdx` (#3899)
withastro/docs: src/content/docs/pt-br/reference/configuration-reference.mdx

**Veja Também:**
- build.server

### cacheDir

<p>

**Tipo:** `string`<br />
**Padrão:** `"./node_modules/.astro"`
</p>

> Define a pasta para deixar em cache artefatos do build. Arquivos nessa pasta vão ser usados em subsequentes builds para acelerar o tempo de build.

Author: Talvez não seria melhor usar "de", como está lá no glossário?

> Define a pasta para deixar em cache artefatos do build. Arquivos nessa pasta vão ser usados em subsequentes builds para acelerar o tempo de build.

Author: Mas agora que vi que em diversos lugares utilizam o "da".

> Define a pasta para deixar em cache artefatos do build. Arquivos nessa pasta vão ser usados em subsequentes builds para acelerar o tempo de build.

Reviewer: Sim, há o uso válido de ambos "de" e "da", o motivo para "da" é que utilizamos a tradução para definir o gênero (a construção), mas também é válido "de" por ser neutro em artigo.

O valor pode ser tanto um caminho absoluto no sistema de arquivos ou um caminho relativo a raiz do projeto.

```js
{
  cacheDir: './minha-pasta-de-cache-personalizada'
}
```

### redirects

<p>

**Tipo:** `Record.<string, RedirectConfig>`<br />
**Padrão:** `{}`<br />
<Since v="2.9.0" />
</p>

Especifica um mapeamento de redirecionamentos onde a chave é a rota a corresponder
e o valor é o caminho pra onde será redirecionado.

Você pode redirecionar tanto rotas estáticas como dinâmicas, mas somente o mesmo tipo de rota.
Por exemplo, você não pode ter um redirecionamento `'/artigo': '/blog/[...slug]'`.

```js
{
  redirects: {
    '/antigo': '/novo',
    '/blog/[...slug]': '/artigos/[...slug]',
  }
}
```

> Para sites estaticamente gerados sem nenhum adapter instalado, isso vai produzir um redirecionamento no cliente usando uma [tag `<meta http-equiv="refresh">`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/meta#http-equiv) e não suporta códigos de status.

Author: Eu fiquei de verificar os links do MDN, mas iria checar de todos os arquivos, por isso acabei não checando esse

> Quando estiver usando SSR ou um adapter estático no modo `output: static`,
> códigos de status não são suportados.
> O Astro irá server requisições GET redirecionadas com um status de `301`

Reviewer: ```suggestion
Quando estiver usando SSR ou um adaptador estático no modo `output: static`,
códigos de status não são suportados.
O Astro irá servir requisições GET redirecionadas com um status de `301`
```

e usar um status de `308` para qualquer outro método de requisição.

> Você pode customizar o [codigo de status de redirecionamento](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#redirection_messages) usando um objeto na configuração de redirecionamento:

Reviewer: Versão em português da página
```suggestion
Você pode customizar o [codigo de status de redirecionamento](https://developer.mozilla.org/pt-BR/docs/Web/HTTP/Status#mensagens_de_redirecionamento) usando um objeto na configuração de redirecionamento:
```

**Veja Também:**
- build.format

### scopedStyleStrategy

<p>

**Tipo:** `'where' | 'class'`<br />
**Padrão:** `'where'`<br />
<Since v="2.4" />
</p>

Especifica a estratégia usada para definir o escopo de estilos dentro de componentes do Astro. Escolha entre:

>   - `'where'` - Usar seletores `:where`, não causando nenhum aumento de especificidade.
>   - `'class'` - Usar seletores baseados em classe, causando um aumento de +1 de especificidade.

Reviewer: "Usa" seria o tempo verbo mais adequado aqui
```suggestion
  - `'where'` - Usa seletores `:where`, não causando nenhum aumento de especificidade.
  - `'class'` - Usa seletores baseados em classe, causando um aumento de +1 de especificidade.
```

> Usar `'class'` é útil quando você quer garantir que os seletores de elementos dentro do Astro sobrescrevam os padrões de estilo globais (e.x. de uma folha de estilo global).

Reviewer: No original é mencionado "components", que acabou faltando aqui.
```suggestion
Usar `'class'` é útil quando você quer garantir que os seletores de elementos em componentes Astro sobrescrevam os padrões de estilo globais (e.x. de uma folha de estilo global).
```

**Padrão:** `'./dist/client'`
</p>

Controla o diretório de saída do seu CSS e JavaScript do lado do cliente apenas quando `output: 'server'` ou `output: 'hybrid'`.
`outDir` controla onde o código é construído.

Esse valor é relativo ao `outDir`.

```js
{

>   output: 'server', // or 'hybrid'

Reviewer: ```suggestion
  output: 'server', // ou 'hybrid'
```

}
```

### build.redirects

<p>

**Tipo:** `boolean`<br />
**Padrão:** `true`<br />
<Since v="2.6.0" />
</p>

> Espeficica se os redirecionamentos são enviados como HTML durante o build.
> Essa opção só se aplica para o modo `output: 'static'`; em SRR os redirecionamentos

Reviewer: ```suggestion
Espeficica se os redirecionamentos são enviados como HTML durante a build.
Essa opção só se aplica para o modo `output: 'static'`; em SSR os redirecionamentos
```

são tratados da mesma forma que todas as respostas.

Essa opção é principalmente para ser usada por adaptadores que possuem arquivos

> de configuração especiais para redirecionamento a não precisa/querem redirecionamentos baseados em HTML.

Reviewer: Errinho de digitação
```suggestion
de configuração especiais para redirecionamentos e não precisam/querem redirecionamentos baseados em HTML.
```

**Padrão:** `never`<br />
<Since v="2.6.0" />
</p>

Controla se os estilos são enviados para o navegador em um arquivo CSS separado ou em linha em tags `<style>`. Escolha entre as seguintes opções:
 - `'always'` - todos os estilos são embutidos em tags `<style>`
 - `'auto'` - apenas folhas de estilo menores que `ViteConfig.build.assetsInlineLimit` (padrão: 4kb) são embutidas. Caso contrário, estilos são enviados em folhas de estilo externas.
 - `'never'` - todos os estilos são enviados em folhas de estilo externas

```js
{
	build: {
		inlineStylesheets: `auto`,
	},
}
```

### build.split

<p>

**Tipo:** `boolean`<br />
**Padrão:** `false`<br />
<Since v="2.7.0" />
</p>

Define como o código SSR deve ser empacotado quando feito build.

Quando `split` é `true`, o Astro vai emitir um arquivo para cada página.
Cada arquivo emitido vai renderizar apenas uma página. As páginas serão emitidas
dentro de uma pasta `dist/pages/`, e os arquivos emitidos vão manter o mesmo caminho de arquivos
da pasta `src/pages`.

```js
{
  build: {
    split: true
  }
}
```

### build.excludeMiddleware

<p>

**Tipo:** `boolean`<br />
**Padrão:** `false`<br />
<Since v="2.8.0" />
</p>

> Define se algum código de middleware SSR vai ser empacotado ou não quando o build for feito.
> 
> Quando habilitado, código de middleware não é empacotado e importado por todas as páginas durante o build. Para ao invés disso executar e importar códigos de middleware manualmente, defina `build.excludeMiddleware: true`:

Reviewer: ```suggestion
Define se algum código de middleware SSR vai ser empacotado ou não quando a build for feita.

Quando habilitado, código de middleware não é empacotado e importado por todas as páginas durante a build. Para ao invés disso, executar e importar códigos de middleware manualmente, defina `build.excludeMiddleware: true`:
```

		assets: true,
	},
}
```

### experimental.viewTransitions

<p>

**Tipo:** `boolean`<br />
**Padrão:** `false`<br />
<Since v="2.9.0" />
</p>

Habilita o suporte experimental para o componente `<ViewTransitions / >`. Com isso habilitado

> você pode adotar a [transições de visualização](/pt-br/guides/view-transitions/) por página usando esse componente
> e habilita animações com a diretiva `transition:animate`.

Reviewer: ```suggestion
você pode optar por utilizar [transições de visualização](/pt-br/guides/view-transitions/) por página usando esse componente
e habilitar animações com a diretiva `transition:animate`.
```
